### PR TITLE
Make sure mark_as_installed respects system config

### PR DESCRIPTION
**Pull request by amaslenn**
## Summary
When system's `.cache_docker_images_locally` is not set to `true`, `_install_path` shouldn't be set and URL should be used instead.

## Test Plan
CI (extended)

## Additional Notes
—

### CODE REVIEW

**Review by TaekyungHeo — APPROVED** [review]

File: tests/test_slurm_installer.py [hunk 1]
```diff
@@ -485,3 +485,16 @@ def test_mark_as_installed(slurm_system: SlurmSystem):
     assert res.success
     assert docker.installed_path == slurm_system.install_path / docker.cache_filename
     assert py_script.git_repo.installed_path == slurm_system.install_path / py_script.git_repo.repo_name
+
+
+@pytest.mark.parametrize("cache", [True, False])
+def test_mark_as_installed_docker_image_system_is_respected(slurm_system: SlurmSystem, cache: bool):
```
Comment (srivatsankrishnan) [inline tests/test_slurm_installer.py:491]: Add more test cases for edge cases

What if DockerImage is already marked as installed?

File: tests/test_base_installer.py [hunk 2]
```diff
@@ -238,3 +238,26 @@ def test_both_is_installed(self, installer: SlurmInstaller):
         assert f1._installed_path is not None, "First file is not installed"
         assert f2._installed_path is not None, "Second file is not installed"
         assert f1._installed_path == f2._installed_path, "Files are installed to different paths"
+
+    def test_order_of_items_does_not_matter(self, installer: SlurmInstaller):
```
[the remark this answers is not in the document]
Comment (amaslenn) [reply tests/test_base_installer.py:242]: IMO parametrization looks confusing here:
```py
    @pytest.mark.parametrize(
        "files",
        [
            [File(src=Path(__file__)), File(src=Path(__file__))],
            [File(src=Path(__file__)), File(src=Path(__file__))],
        ],
    )
```
As actual file is the same, two different parameters look identical. Splitting them into `f1: File, f2: File` also doesn't make sense as the code of the test will remain the same as it is now, with both `[f1, f2]` followed by `[f2, f1]` cases.

Let me know if you see how to improve this test.

**Review by srinivas212 — APPROVED** [review]
